x-pack/plugin/monitoring/src/main/java/org/elasticsearch/xpack/monitoring/exporter/Exporter.java

```diff
@@ -102,9 +102,26 @@ public void validate(final String value, final Map<Setting<?>, Object> settings)
     /**
      * Every {@code Exporter} allows users to use a different index time format.
      */
-    private static final Setting.AffixSetting<String> INDEX_NAME_TIME_FORMAT_SETTING =
+    static final Setting.AffixSetting<String> INDEX_NAME_TIME_FORMAT_SETTING =
             Setting.affixKeySetting("xpack.monitoring.exporters.","index.name.time_format",
-                    key -> Setting.simpleString(key, Property.Dynamic, Property.NodeScope));
+                    key -> Setting.simpleString(
+                        key,
+                        Exporter.INDEX_FORMAT,
+                        new Setting.Validator<String>() {
+                            @Override
+                            public void validate(String value) {
+                                try {
+                                    if (value != null) {
+                                        DateFormatter.forPattern(value).withZone(ZoneOffset.UTC);
```

Comment:
Is the generic type for this AffixSetting wrong? It seems like if we are going to parse into a DateFormatter, we should be doing that once, not duplicating here and in Exporter.dateTimeFormatter (where it also looks like we have default value logic?).

Reply:
@rjernst, it is true that a `DateFormatter` instance is ultimately what is produced for this setting. I could change `INDEX_NAME_TIME_FORMAT_SETTING` from its original `AffixSetting<String>` to a `AffixSetting<DateFormatter>` if that would be preferable.

Reply:
I think that makes the most sense. In general, a major goal of the settings infrastructure is to do type parsing and validation for the intended type. In this case it seems we are adding checking, but the parsing isn't actually happening within the Setting instance.